go/auth/token.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+//
+// Code used to support authentication tokens for arbitrary purposes.
+//
+package auth
+
+import (
+	"bytes"
+	"encoding/json"
+	"math"
+	"time"
+
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+const CurrentTokenVersion = 1
+
+type TokenKey struct {
+	UID      keybase1.UID             `json:"uid"`
+	Username libkb.NormalizedUsername `json:"username"`
+	KID      keybase1.KID             `json:"kid"`
+}
+
+type TokenBody struct {
+	Key     TokenKey `json:"key"`
+	Type    string   `json:"type"`
+	Version int      `json:"version"`
+}
+
+type TokenClient struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+type Token struct {
+	Body         TokenBody   `json:"body"`
+	Client       TokenClient `json:"client"`
+	CreationTime int64       `json:"ctime"`
+	ExpireIn     int         `json:"expire_in"`
+	Tag          string      `json:"tag"`
+}
+
+func NewToken(uid keybase1.UID, username libkb.NormalizedUsername, kid keybase1.KID,
+	tokenType string, expireIn int, clientName string, clientVersion string) *Token {
+	return &Token{
+		Body: TokenBody{
+			Key: TokenKey{
+				UID:      uid,
+				Username: username,
+				KID:      kid,
+			},
+			Type:    tokenType,
+			Version: CurrentTokenVersion,
+		},
+		Client: TokenClient{
+			Name:    clientName,
+			Version: clientVersion,
+		},
+		CreationTime: time.Now().Unix(),
+		ExpireIn:     expireIn,
+		Tag:          "signature",
+	}
+}
+
+func (t Token) Bytes() []byte {
+	bytes, err := json.Marshal(&t)
```

Comment:
Yeah, sorry about that. I opened some libkb files and saw all the jsonw stuff and thought that's what we wanted for some reason.